src/assets/specs/command-api.yaml

```diff
@@ -0,0 +1,108 @@
+openapi: "3.0.0"
+info:
+  title: "Rancher Desktop API"
+  version: 0.0.1
+paths:
+  /v0/list-settings:
+    get:
+      operationId: listSettings
+      summary:  List the current preference settings
+      responses:
+        '200':
+          description: |-
+            200 response
+          content:
+            application/json:
+              schema:
+                "$ref" : "#/components/schemas/preferences"
+
+  /v0/set:
+    put:
+      operationId: updateSettings
+      summary:  Updates the specified preference settings
+      requestBody:
+        description: |-
+          JSON block consisting of some or all of the current preferences,
+          with changes applied to any number of settings the backend supports changing this way.
+        content:
+          application/json:
+            schema:
+              "$ref" : "#/components/schemas/preferences"
+        required: true
+      responses:
+        '202':
+          description: |-
+            202 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+        '400':
+          description: |-
+            400 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+
+  /v0/shutdown:
+    put:
+      operationId: shutdownApp
+      summary:  Shuts down Rancher Desktop
+      responses:
+        '202':
+          description: |-
+            202 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+components:
+  schemas:
+    preferences:
```

Comment:
Hmm, do we need to auto-generate this part? Because we're dumping the raw config JSON over the wire, we'd need to keep things in sync.

(Also, we internally call this `settings`; the different name is fine, just curious.)

Reply:
If you're thinking of doing something like generating the type spec from the default values in `settings.ts`, I could go along with that.

Reply:
Yeah, exactly, but that doesn't sound easy at all :(